pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -454,13 +465,14 @@ func (e *EndpointController) syncService(key string) error {
 // some stragglers could have been left behind if the endpoint controller
 // reboots).
 func (e *EndpointController) checkLeftoverEndpoints() {
-	list, err := e.client.Core().Endpoints(metav1.NamespaceAll).List(metav1.ListOptions{})
+	//list, err := e.client.Core().Endpoints(metav1.NamespaceAll).List(metav1.ListOptions{})
+	list, err := e.endpointsLister.List(labels.Everything())
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("Unable to list endpoints (%v); orphaned endpoints will not be cleaned up. (They're pretty harmless, but you can restart this component if you want another attempt made.)", err))
 		return
 	}
-	for i := range list.Items {
-		ep := &list.Items[i]
+	for i := range list {
+		ep := list[i]
```

Comment:
- [ ] Why not use the two-value `range` clause here, given that [`EndpointsLister.List` returns a slice of pointers](https://github.com/kubernetes/client-go/blob/master/listers/core/v1/endpoints.go#L31)?  
```go
for _, ep := range list {
```